Quantum-Control-Applications/Superconducting/Single Flux Tunable Transmon/SSB mixer for readout/macros.py

```diff
@@ -72,49 +90,47 @@ def active_reset_until_success(threshold, max_tries=1, Ig=None):
             dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", Ig),
         )
         # Play a pi pulse to get back to the ground state
-        play("pi", "qubit", condition=(Ig < threshold))
+        play("pi", "qubit", condition=(Ig > threshold))
         # Increment the number of tries
         assign(counter, counter + 1)
     return Ig, counter
 
 
-# Macro for measuring the ground and excited states with single shot
-def ge_singleshot_measurement(cooldown_time):
-    """Macro measuring the singleshot qubit's ground and excited states.
-
-    :param cooldown_time: cooldown time between two successive qubit state measurements in clock cycle unit (4ns).
-    :return: singleshot I and Q data for the ground and excited states respectively [Ig, Qg, Ie, Qe].
+# Macro for measuring the qubit state with single shot
+def singleshot_measurement(threshold=None, state=None, I=None, Q=None):
```

Comment:
Maybe `readout macro`? Because it also supports not single-shot...

Reply:
Well this is a terminology thing, for me singleshot means that we only measure the state once no matter the fidelity, as opposed to the other readout macro with averages

Reply:
I see, then maybe `single_measurement`?
I think `singleshot` usually means it also gives you the result with high fidelity.

Reply:
Oh ok I didn't know, I'll change it to single_measurement :)